### PR TITLE
Internal: Editor V2 - Publish button extensibility [ED-9730]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: packages/packages/documents-ui/src/components/top-bar/primary-action-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+			transformOrigin={ {
+				vertical: 'top',
+				horizontal: 'right',
+			} }
+			PaperProps={ {
+				sx: { mt: 2, ml: 3 },
+			} }
+		>
+			{ saveActions.map( ( { MenuItem, id } ) => ( [
+				<Divider key={ `${ id }-divider` } orientation="horizontal" />,
```
Comment (rotemee) [inline packages/packages/documents-ui/src/components/top-bar/primary-action-menu.tsx:39]: Remove first Divider.

File: packages/packages/documents-ui/src/components/top-bar/primary-action-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+				<MenuItem key={ id } />,
+			] ) ) }
+
+			{ defaultActions.length > 0 && <Divider orientation="horizontal" /> }
```
Comment (rotemee) [inline packages/packages/documents-ui/src/components/top-bar/primary-action-menu.tsx:43]: Check if "horizontal" is the default.

File: packages/packages/documents-ui/src/components/top-bar/primary-action.tsx [hunk 3]
```diff
@@ -17,19 +35,34 @@ export default function PrimaryAction() {
...
+							},
+						},
+					} }
+				>
+					<Button sx={ { px: 0 } } { ...bindTrigger( popupState ) }>
+						<ChevronDownIcon />
+					</Button>
+				</Tooltip>
+			</ButtonGroup>
+			<PrimaryActionMenu { ...bindMenu( popupState ) } anchorEl={ ref.current } />
```
Comment (rotemee) [inline packages/packages/documents-ui/src/components/top-bar/primary-action.tsx:64]: Remove anchorEl.

File: packages/packages/documents-ui/src/hooks/use-document-save-draft-props.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+export default function useDocumentSaveDraftProps(): ActionProps {
+	const document = useActiveDocument();
+	const { saveDraft } = useActiveDocumentActions();
+
+	return {
+		icon: FileReportIcon,
+		title: __( 'Save Draft', 'elementor' ),
+		onClick: saveDraft,
+		disabled: ! document || document.isSavingDraft || ! document.isDirty,
```
Comment (rotemee) [inline packages/packages/documents-ui/src/hooks/use-document-save-draft-props.ts:14]: add `document.isSaving`

File: packages/packages/documents-ui/src/components/top-bar/__tests__/primary-action-menu.test.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+		expect( menuItems ).toHaveLength( 2 );
+		expect( menuItems[ 0 ] ).toHaveTextContent( 'First action' );
+		expect( menuItems[ 1 ] ).toHaveTextContent( 'Second action' );
+
+		// Act.
+		menuItems[ 0 ].click();
+
+		// Assert.
+		expect( onClick ).toHaveBeenCalledTimes( 1 );
```
Comment (rotemee) [inline packages/packages/documents-ui/src/components/top-bar/__tests__/primary-action-menu.test.tsx:51]: Remove.

File: packages/packages/documents-ui/src/hooks/__tests__/use-document-save-draft-props.test.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+		// Act.
+		const { result } = renderHook( () => useDocumentSaveDraftProps() );
+		result.current.onClick?.();
+
+		// Assert.
+		expect( documentActions.saveDraft ).toBeCalledTimes( 1 );
+	} );
+
+	it.each( [
+		{
```
Comment (rotemee) [inline packages/packages/documents-ui/src/hooks/__tests__/use-document-save-draft-props.test.ts:32]: isSaving.

**Review by rotemee — CHANGES_REQUESTED** [review]

File: packages/packages/documents-ui/src/components/top-bar/primary-action-menu.tsx (hunk 1 above)
Comment (StyleShit) [reply packages/packages/documents-ui/src/components/top-bar/primary-action-menu.tsx:39]: Done

File: packages/packages/documents-ui/src/components/top-bar/primary-action-menu.tsx (hunk 2 above)
Comment (StyleShit) [reply packages/packages/documents-ui/src/components/top-bar/primary-action-menu.tsx:43]: Done

File: packages/packages/documents-ui/src/components/top-bar/primary-action.tsx (hunk 3 above)
Comment (StyleShit) [reply packages/packages/documents-ui/src/components/top-bar/primary-action.tsx:64]: Done

File: packages/packages/documents-ui/src/hooks/use-document-save-draft-props.ts (hunk 4 above)
Comment (StyleShit) [reply packages/packages/documents-ui/src/hooks/use-document-save-draft-props.ts:14]: Done

File: packages/packages/documents-ui/src/components/top-bar/__tests__/primary-action-menu.test.tsx (hunk 5 above)
Comment (StyleShit) [reply packages/packages/documents-ui/src/components/top-bar/__tests__/primary-action-menu.test.tsx:51]: Done

**Review by rotemee — APPROVED** [review]

**Pull request merged**